pkgdb/src/ld-floxlib.c

```diff
@@ -97,54 +110,110 @@ la_objsearch( const char * name, uintptr_t * cookie, unsigned int flag )
       if ( fd != -1 ) { close( fd ); }
       else
         {
-          char * basename          = strrchr( name, '/' );
-          char * flox_env_lib_dirs = getenv( "FLOX_ENV_LIB_DIRS" );
-
+          char * basename = strrchr( name, '/' );
           if ( basename != NULL ) { basename++; }
           else { basename = (char *) name; }
 
-          if ( flox_env_lib_dirs != NULL )
+          if ( flox_env_lib_dirs_count == -1 )
             {
-              // Iterate over the colon-separated list of paths in
-              // FLOX_ENV_LIB_DIRS looking for the requested library.
-              // If found, return the full path to the library and
-              // otherwise return the original name.
-              char * flox_env_library_dir = NULL;
-              flox_env_library_dir        = strtok( flox_env_lib_dirs, ":" );
-              while ( flox_env_library_dir != NULL )
+              // Copy the contents of the FLOX_ENV_LIB_DIRS variable into
+              // flox_env_lib_dirs_buf and tokenize the buffer by replacing
+              // colons with NULLs as we count the entries, saving pointers
+              // to each of the paths in the flox_env_lib_dirs[] array.
+              flox_env_lib_dirs_count = 0;
+              const char * flox_env_lib_dirs_env
+                = getenv( "FLOX_ENV_LIB_DIRS" );
+              if ( flox_env_lib_dirs_env != NULL )
                 {
-                  (void) snprintf( name_buf,
-                                   sizeof( name_buf ),
-                                   "%s/%s",
-                                   flox_env_library_dir,
-                                   basename );
-                  if ( debug_ld_floxlib )
+                  if ( sizeof( flox_env_lib_dirs_env )
+                       > FLOX_ENV_LIB_DIRS_MAXLEN )
```

Comment:
```suggestion
                       >= FLOX_ENV_LIB_DIRS_MAXLEN )
```
For the weird case of of them being equal, there is no terminating `\0` written.